src/codec.rs

```diff
@@ -131,6 +131,17 @@ pub trait Input {
 		self.read(&mut buf[..])?;
 		Ok(buf[0])
 	}
+
+	/// Descend into nested reference when decoding.
+	/// This is called when decoding a new refence-based instance,
+	/// such as `Vec` or `Box`
+	fn descend(&mut self) -> Result<(), Error> {
```

Comment:
```suggestion
	fn descend_heap(&mut self) -> Result<(), Error> {
```

Also not that good, but only `descend` isn't enough IMHO.